File: src/LightNap.Core/Identity/Models/LoginSuccessType.cs
namespace LightNap.Core.Identity.Models
{
    /// <summary>
    /// Represents the result of a successful login operation.
    /// </summary>
    public enum LoginSuccessType
    {
        /// <summary>
        /// Indicates the response contains an access token.
        /// </summary>
        AccessToken,

        /// <summary>
        /// Indicates the user must verify their email address before they can get an access token.
        /// </summary>
        EmailVerificationRequired,

        /// <summary>
        /// Indicates the user must submit the 2FA code before they can get an access token.
        /// </summary>
        TwoFactorRequired
    }
}

File: src/lightnap-ng/src/app/core/backend-api/dtos/identity/response/login-success-type.ts
/**
 * Represents the possible outcomes of a successful login attempt.
 *
 * @typedef {("AccessToken" | "EmailVerificationRequired" | "TwoFactorRequired")} LoginSuccessType
 *
 * @property {"AccessToken"} AccessToken - Indicates that the login was successful and an access token was provided.
 * @property {"EmailVerificationRequired"} EmailVerificationRequired - Indicates that the login was successful but email verification is required.
 * @property {"TwoFactorRequired"} TwoFactorRequired - Indicates that the login was successful but two-factor authentication is required.
 */
export type LoginSuccessType = "AccessToken" | "EmailVerificationRequired" | "TwoFactorRequired";
